src/fstab-generator/fstab-generator.c

```diff
@@ -634,6 +638,7 @@ static int parse_fstab(bool initrd) {
 
                 makefs = fstab_test_option(me->mnt_opts, "x-systemd.makefs\0");
                 growfs = fstab_test_option(me->mnt_opts, "x-systemd.growfs\0");
+                rwonly = fstab_test_option(me->mnt_opts, "x-systemd.rw-only\0");
```

Comment:
Hmm, I'm not convinced by the dash in the name. I think it'll be confusing to have all the other terms without a dash, except for this one.